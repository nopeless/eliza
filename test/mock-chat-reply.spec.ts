import { client } from "./fixtures";

describe(`translate`, () => {
  test(`eliza tl|translate`, async () => {
    const m = await client.sendExpect(
      `eliza tl`,
      /missing translation context/i
    );

    expect(m).to.matchSnapshot();
  });
  test(`translate syntax`, async () => {
    const m = await client.sendExpect(
      `eliza japan translate poco loco`,
      /translated.+spanish/i
    );

    expect(m).to.matchSnapshot();
  });
  test(`tl shorthand`, async () => {
    const m = await client.sendExpect(
      `Eliza en tl bonjour, monsieur`,
      /hello|good|morning/i
    );

    expect(m).to.matchSnapshot();
  });
  test(`natural syntax`, async () => {
    const m = await client.sendExpect(
      `eliza translate to japanese: poco loco`,
      /translated/i
    );

    expect(m).to.matchSnapshot();
  });
  test(`no colon required`, async () => {
    const m = await client.sendExpect(
      `eliza translate poco loco`,
      /translated/i
    );

    expect(m).to.matchSnapshot();
  });
  test(`forgot colon`, async () => {
    const m = await client.sendExpect(
      `eliza translate japanese poco loco`,
      /forg[eo]t/i
    );

    expect(m).to.matchSnapshot();
  });
});

describe.skipIf(!process.env.RAPID_API_KEY)(`define`, () => {
  test(`missing word`, async () => {
    const m = await client.sendExpect(`eliza define`, /specify/i);

    expect(m).to.matchSnapshot();
  });
  test(`proper`, async () => {
    const m = await client.sendExpect(`eliza define air`, /definition/i);

    expect(m).to.matchSnapshot();
  });
});

describe(`hi`, () => {
  test(`hi`, async () => {
    const m = await client.sendExpect(
      `eliza hello`,
      /hello|hi|up|doing|good|hru|how/i
    );

    console.log(m);
  });
});

describe(`url unshorten`, () => {
  test(`rickroll`, async () => {
    const m = await client.sendExpect(`http://tiny.cc/mtd7vz`, /dQw4w9WgXcQ/);

    console.log(m);
  });
});

describe(`help`, () => {
  test(`help nothing`, async () => {
    const m = await client.sendExpect(`eliza help`, /help with/i);

    console.log(m);
  });
  test(`help correct match`, async () => {
    const m = await client.sendExpect(`eliza help translate`, /language/i);

    expect(m).to.matchSnapshot();
  });
  test(`help vague match`, async () => {
    const m = await client.sendExpect(`eliza help translte it`, /mean/i);

    expect(m).to.matchSnapshot();
  });
  test(`help not a command help`, async () => {
    const m = await client.sendExpect(
      `eliza help I need some guava juice but I can't find them in my store`,
      /I would/i
    );

    console.log(m);
  });
});
