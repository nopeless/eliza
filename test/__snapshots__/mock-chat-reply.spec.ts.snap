// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`define > missing word 1`] = `"Please specify a word to define. ex) define hello"`;

exports[`define > proper 1`] = `
"Here is the definition of 'air'
expose to fresh air"
`;

exports[`help > help correct match 1`] = `
"Here is the description
translate text to another language"
`;

exports[`help > help vague match 1`] = `"Did you mean 'translate'? Try \\"help translate\\""`;

exports[`translate > eliza tl|translate 1`] = `"missing translation context. example: \\"tl こんにちは世界\\""`;

exports[`translate > forgot colon 1`] = `
"translated from Spanish; Castilian to English:
\\"little crazy japanese\\"
> 'japanese' seems like a language name. Did you forget a colon? ex) translate japanese: poco loco"
`;

exports[`translate > natural syntax 1`] = `
"translated from Spanish; Castilian to Japanese:
\\"ちょっとおかしい\\""
`;

exports[`translate > no colon required 1`] = `
"translated from Spanish; Castilian to English:
\\"little crazy\\""
`;

exports[`translate > tl shorthand 1`] = `
"translated from French to English:
\\"Good morning sir\\""
`;

exports[`translate > tl shorthand 2`] = `
"translated from French to English:
\\"Good morning sir\\""
`;

exports[`translate > translate syntax 1`] = `
"translated from Spanish; Castilian to Japanese:
\\"ちょっとおかしい\\""
`;
